src/nvim/strings.c

```diff
@@ -45,6 +45,41 @@
 #include "nvim/os/os.h"
 #include "nvim/os/shell.h"
 
+/// Compares two strings for equality.
+bool strequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcmp(a, b) == 0;
+}
+
+/// Compares two sized strings for equality.
+bool strnequal(const char *a, const char *b, size_t size)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strncmp(a, b, size) == 0;
+}
+
+/// Compares two strings for equality ignoring differences in case.
+bool striequal(const char *a, const char *b)
+  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_PURE
+{
+  return strcasecmp(a, b) == 0;
```

Comment:
@aktau It makes sense since current implementation of `vim_stricmp` is actually not correct:

1. it is not correct to compare unicode strings just by lowercasing them;
2. they work only for 8-bit encodings.